element-hq/element-x-android
# Categorise members by role in the ChangeRoles screen

## Type of change

- [x] Other : UI change.

## Content

- Categorises members in different sections by role in the change roles screens.
- For the change role to moderator screen, also add a subtitle notice confirming administrators already have moderator permissions.
- For users who have been invited to the room, add a 'Pending' label next to their names.
- Automatically reload member list when power levels or membership of members change.
- Replace empty space next to users whose level can't be changed to disabled checkboxes to match [the latest Figma designs](https://www.figma.com/file/A2pAEvTEpJZBiOPUlcMnKi/Settings-%2B-Room-Details-(new)?type=design&node-id=1739-17431&mode=design&t=jsOy8TlfwaXCU1dR-4).

## Motivation and context

Fixes #2593 .

## Screenshots / GIFs

|Before|After|
|-|-|
|![](https://media.githubusercontent.com/media/element-hq/element-x-android/103743c33812a7cd4dae14f7104ad0a234f48c52/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.roomdetails.impl.rolesandpermissions.changeroles_ChangeRolesView_null_ChangeRolesView-Day-9_10_null_3%2CNEXUS_5%2C1.0%2Cen%5D.png)|![](https://media.githubusercontent.com/media/element-hq/element-x-android/c6ea6d2327a0532fdb74ba30d474217390fea526/tests/uitests/src/test/snapshots/images/ui_S_t%5Bf.roomdetails.impl.rolesandpermissions.changeroles_ChangeRolesView_null_ChangeRolesView-Day-9_10_null_3%2CNEXUS_5%2C1.0%2Cen%5D.png)|

## Tests



- Go into a room where you're an admin.
- Open its settings -> roles and permissions -> roles (admins or moderators).
- Verify the new design is used and the feature works.

## Tested devices

- OS version(s): 14

## Checklist
